netaddr.go

```diff
@@ -291,11 +253,11 @@ func (ip IP) BitLen() uint8 {
 
 // Zone returns ip's IPv6 scoped addressing zone, if any.
 func (ip IP) Zone() string {
-	if ip.z == nil {
+	switch ip.z {
```

Comment:
Revert this part? It's unrelated to this change and is just a stylistic preference change as far as I can tell. Can discuss on its own, but not worth mixing it with this change.

Reply:
It wasn't stylistic, it was an attempt at avoiding a type assertion in the common case where the IP has no zone.

That said, looking at the SSA, the compiler optimized the prior implementation much better than I expected, so there's no benefit to doing this. Reverted.